server/protocols/jmap/src/main/java/org/apache/james/jmap/JMAPConfiguration.java

```diff
@@ -0,0 +1,106 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.jmap;
+
+import java.util.Optional;
+
+import org.apache.james.util.Port;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+
+public class JMAPConfiguration {
+
+    // TODO move tests too
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private Optional<Boolean> enabled = Optional.empty();
+        private Optional<Boolean> wiretap = Optional.empty();
+        private Optional<Port> port = Optional.empty();
+
+        private Builder() {
+
+        }
+
+        public Builder enabled(boolean enabled) {
+            this.enabled = Optional.of(enabled);
+            return this;
+        }
+
+        public Builder wiretap() {
+            return wiretap(true);
+        }
+
+        public Builder wiretap(boolean enabled) {
+            this.wiretap = Optional.of(enabled);
+            return this;
+        }
+
+        public Builder enable() {
+            return enabled(true);
+        }
+
+        public Builder disable() {
+            return enabled(false);
+        }
+
+        public Builder port(Port port) {
+            this.port = Optional.of(port);
+            return this;
+        }
+
+        public Builder randomPort() {
+            this.port = Optional.empty();
+            return this;
+        }
+
+        public JMAPConfiguration build() {
+            Preconditions.checkState(enabled.isPresent(), "You should specify if JMAP server should be started");
```

Comment:
can `enabled` default to true?

Reply:
It was not until now, why do you think it should? Even more as long as we don't finish the dev of it?

Reply:
I'm expecting an user can start his James when forgetting to put `enable=true` in `jmap.properteis`. We're encouraging default configurations right?

Reply:
I believe it's a port for now. Let's not start changing everything around. If you think it's better though, I invite you to open a PR about it :)

Reply:
No default value changes please